packages/webcomponents/src/components/business-forms/payment-provisioning/document-uploads/business-document-upload-form-step.tsx

```diff
@@ -0,0 +1,236 @@
+  @Event() serverError: EventEmitter<DocumentFormServerErrorEvent>;
+
+  private api: any;
+
+  get businessEndpoint() {
+    return `entities/business/${this.businessId}`;
+  }
+
+  get documentEndpoint() {
+    return 'entities/document';
+  }
+
+  componentWillLoad() {
+    const missingAuthTokenMessage = 'Warning: Missing auth-token. The form will not be functional without it.';
+    const missingBusinessIdMessage = 'Warning: Missing business-id. The form requires an existing business-id to function.';
+    if (!this.authToken) console.error(missingAuthTokenMessage);
+    if (!this.businessId) console.error(missingBusinessIdMessage);
+
+    this.formController = new FormController(businessDocumentSchema(this.allowOptionalFields));
+    this.api = Api({ authToken: this.authToken, apiOrigin: config.proxyApiOrigin });
+    this.fetchData();
+  }
+
+  componentDidLoad() {
+    this.formController.values.subscribe(values =>
+      this.documents = { ...values }
+    );
+    this.formController.errors.subscribe(
+      errors => (this.errors = { ...errors }),
+    );
+  }
+
+  inputHandler = (name: string, value: string) => {
+    this.formController.setValues({
+      ...this.formController.values.getValue(),
+      [name]: value,
+    });
+  }
+
+  private fetchData = async () => {
+    this.formLoading.emit(true);
+    try {
+      const response: IApiResponse<IBusiness> = await this.api.get(this.businessEndpoint);
+      this.documents = response.data.documents;
+    } catch (error) {
+      this.serverError.emit({ data: error, message: DocumentFormServerErrors.fetchData });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+
+  createDocumentRecord = async (docData: EntityDocument) => {
+    this.formLoading.emit(true);
+    const payload = docData.record_data;
+    try {
+      const response = await this.api.post(this.documentEndpoint, JSON.stringify(payload));
+      return this.handleDocRecordResponse(docData, response);
+    } catch (error) {
+      this.serverError.emit({ data: error, message: DocumentFormServerErrors.sendData });
+      return false;
+    }
+  }
+
+  handleDocRecordResponse = (docData: EntityDocument,response: any) => {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: DocumentFormServerErrors.sendData });
+      return false;
+    } else {
+      docData.setPresignedUrl(response.data.presigned_url);
+      return true;
+    }
+  }
+
+  uploadDocument = async (docData: EntityDocument) => {
+    if (!docData.presigned_url) {
+      throw new Error('Presigned URL is not set');
+    }
+
+    const response = await fetch(docData.presigned_url, {
+      method: 'PUT',
+      body: docData.fileString,
+    })
+
+    return this.handleUploadResponse(response);
+  }
+
+  handleUploadResponse = (response: any) => {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: DocumentFormServerErrors.sendData });
```

Comment:
We could either rename this event to `error-event` or add `server-error` to Analytics events list.
`packages/webcomponents/src/api/Analytics.ts:63`

Reply:
hey @TiagoFuelber  - I completely agree with your suggestions about the way we are handling the error event. 

I want to implement these changes in a PR that will follow this one - the way it is currently set up for the PaymentProvisioning component is that the parent component listens for `serverError` to be emitted from its child components (the different form steps) and then uses that to display an error on the page. 

I want to move in this direction and get the `error-event` emitted from the component at each step, the way you've outlined here, but to do so I will also need to make some refactors to all the other form step components. 

I think its a little too much to jam into this PR - so I'd like to do this work in another PR that I will go ahead and start working on 👍🏻